src/bootman/bootman.c

```diff
@@ -446,6 +458,174 @@ bool boot_manager_set_uname(BootManager *self, const char *uname)
+                                      self->initrd_nodep);
+        initrd_source = string_printf("%s/%s",
+                                      self->initrd_nodep_dir,
+                                      self->initrd_nodep);
+        if (!cbm_files_match(initrd_source, initrd_target)) {
+                if (!copy_file_atomic(initrd_source, initrd_target, 00644)) {
+                        LOG_FATAL("Failed to install initrd %s: %s",
+                                  initrd_target,
+                                  strerror(errno));
+                        return;
+                }
+        }
+}
+
+void boot_manager_remove_initrd_nodep(BootManager * self)
+{
+        autofree(char) *base_path = NULL;
+        autofree(char) *initrd_dir = NULL;
+        autofree(char) *initrd_target = NULL;
+        bool is_uefi = ((self->bootloader->get_capabilities(self) & BOOTLOADER_CAP_UEFI) ==
+                        BOOTLOADER_CAP_UEFI);
+        const char *efi_boot_dir =
+            is_uefi ? self->bootloader->get_kernel_destination(self) : NULL;
+        DIR *dir = NULL;
+        struct dirent *ent = NULL;
+        if (!self || !self->initrd_nodep_dir) {
+                return;
+        }
+        /* if it's UEFI, then bootloader->get_kernel_dst() must return a value. */
+        if (is_uefi && !efi_boot_dir) {
+                return;
+        }
+
+        base_path = boot_manager_get_boot_dir((BootManager *)self);
+        initrd_dir = string_printf("%s%s",
+                                      base_path,
+                                      (is_uefi ? efi_boot_dir : ""));
+        dir = opendir(initrd_dir);
+        if (!dir) {
+                LOG_ERROR("Error opening %s: %s", initrd_dir, strerror(errno));
+        }
+
+        while ((ent = readdir(dir)) != NULL) {
+                if (strstr(ent->d_name, "img.gz\0") == NULL) {
+                        continue;
+                }
+                if (self->initrd_nodep == NULL || strcmp(ent->d_name, self->initrd_nodep) != 0) {
+
+                        initrd_target = string_printf("%s%s/%s",
+                                                      base_path,
+                                                      (is_uefi ? efi_boot_dir : ""),
+                                                      ent->d_name);
+                        /* Remove old initrd */
+                        if (nc_file_exists(initrd_target)) {
+                                if (unlink(initrd_target) < 0) {
+                                        LOG_ERROR("Failed to remove legacy-path UEFI initrd %s: %s",
+                                                  initrd_target,
+                                                  strerror(errno));
+                                }
+                        }
+                }
+
+        }
+        closedir(dir);
+}
+
+void boot_manager_write_initrd_nodep(const BootManager *manager, CbmWriter *writer)
```

Comment:
Seems to be redundant with all the extra checks, instead of conditionally calling the function, but just a nitpick